app/V1Module/presenters/ShadowAssignmentsPresenter.php

```diff
@@ -0,0 +1,377 @@
+    }
+
+    // localized texts cannot be empty
+    if (count($req->getPost("localizedTexts")) == 0) {
+      throw new InvalidArgumentException("No entry for localized texts given.");
+    }
+
+    // old values of some attributes
+    $wasPublic = $assignment->isPublic();
+    $isPublic = filter_var($req->getPost("isPublic"), FILTER_VALIDATE_BOOLEAN);
+    $sendNotification = $req->getPost("sendNotification") ? filter_var($req->getPost("sendNotification"), FILTER_VALIDATE_BOOLEAN) : true;
+
+    $assignment->incrementVersion();
+    $assignment->updatedNow();
+    $assignment->setIsPublic($isPublic);
+    $assignment->setIsBonus(filter_var($req->getPost("isBonus"), FILTER_VALIDATE_BOOLEAN));
+    $assignment->setMaxPoints($req->getPost("maxPoints"));
+
+    if ($sendNotification && $wasPublic === false && $isPublic === true) {
+      // assignment is moving from non-public to public, send notification to students
+      $this->assignmentEmailsSender->assignmentCreated($assignment);
+    }
+
+    // go through localizedTexts and construct database entities
+    $localizedTexts = [];
+    foreach ($req->getPost("localizedTexts") as $localization) {
+      $lang = $localization["locale"];
+
+      if (array_key_exists($lang, $localizedTexts)) {
+        throw new InvalidArgumentException("Duplicate entry for language '$lang' in localizedTexts");
+      }
+
+      // create all new localized texts
+      $externalAssignmentLink = trim(Arrays::get($localization, "link", ""));
+      if ($externalAssignmentLink !== "" && !Validators::isUrl($externalAssignmentLink)) {
+        throw new InvalidArgumentException("External assignment link is not a valid URL");
+      }
+
+      $localized = new LocalizedShadowAssignment(
+        $lang, $localization["name"], $localization["text"],
+        $externalAssignmentLink ?: null
+      );
+
+      $localizedTexts[$lang] = $localized;
+    }
+
+    // make changes to database
+    Localizations::updateCollection($assignment->getLocalizedTexts(), $localizedTexts);
+
+    foreach ($assignment->getLocalizedTexts() as $localizedText) {
+      $this->shadowAssignments->persist($localizedText, false);
+    }
+
+    $this->shadowAssignments->flush();
+    $this->sendSuccessResponse($this->shadowAssignmentViewFactory->getAssignment($assignment));
+  }
+
+  /**
+   * Create new shadow assignment in given group.
+   * @POST
+   * @Param(type="post", name="groupId", description="Identifier of the group")
+   * @throws ForbiddenRequestException
+   * @throws BadRequestException
+   * @throws NotFoundException
+   */
+  public function actionCreate() {
+    $req = $this->getRequest();
+    $group = $this->groups->findOrThrow($req->getPost("groupId"));
+
+    if (!$this->groupAcl->canCreateShadowAssignment($group)) {
+      throw new ForbiddenRequestException("You are not allowed to create assignment in given group.");
+    }
```

Comment:
This should go to checkCreate().

Reply:
I thought, that @Teyras wanted to have only checks which uses query parameters in `check` methods, `groupId` is not a query param here.